src/Http/Http/src/DefaultHttpContext.cs

```diff
@@ -44,7 +44,7 @@ public sealed class DefaultHttpContext : HttpContext
         /// Initializes a new instance of the <see cref="DefaultHttpContext"/> class.
         /// </summary>
         public DefaultHttpContext()
-            : this(new FeatureCollection())
+            : this(new FeatureCollection(10))
```

Comment:
Why 10?

Reply:
That's taken from the discussion in the original issue, below usage examples. Can we come up with a better number?

Reply:
Another option would be to add a new constructor to `DefaultHttpContext` as well.

Reply:
10 seems reasonable based on common features used based on this list https://github.com/dotnet/aspnetcore/tree/main/src/Http/Http.Features/src. Add a comment, put it in a constant.

Reply:
Done.